### ISSUE TITLE
Some question from paper and about design mindset!!

### ISSUE THREAD

**Issue by ken881015** [xiaohanghu/2CET-GAN#1]
Hello, author. 
Not long ago, I finished reading your paper. First of all, thank you for your ingenuity and contribution. However, there are several questions that I still cannot understand after reading the paper, and I hope to receive your response here.

### The objective function.
- From my understanding, the three models (E, G, D) used in the paper are jointly trained. 
- However, I find the **training of E** somewhat puzzling. E is influenced by two loss terms: 
  - one is the $L_{cyc_e}$ term from the **ENE** reconstruction, which I understand that you try to assign "expression meaning" to the output of E for G to know what kind of expression should be added on normal face
  - but the $L_{code}$ term in the **NEN** seems a bit odd. I'd like to ask, what is the purpose of having E learn along with a random distribution $\tilde{c}$?
    - The assumption that concerns me is this: In the training process, you only categorize into two classes (with expression and without expression). So, if today I sample a vector from $\tilde{c}$ and hypothetically get a smiling face (just like fig 2 left in paper), I cannot guarantee that adding delta to c will result in smile with slight variation. It's possible that it could result in a crying face or an angry face, right?
    - If I collect 100 smiling face and input to Encoder, will these 100 latent vectors be a cluster in high dimension space?
    - if not, the chapter of continuity analysis in paper is somehow weird to me...

**Comment by xiaohanghu** [comment]
The $L_{code}$ is inspired by the ”info loss” of InfoGAN. We can think of this at a high reasoning level.
$\tilde{c}$ has no meaning at the beginning, as we minimize the $L_{code}$ in the training process, the **G** will gradually give a meaning to $\tilde{c}$, and **E** will learn how to extract this meaning out from the fake image. If **G** continuously ignore $\tilde{c}$, or it does not encode $\tilde{c}$ into the fake image, it is impossible to extract or decode $\tilde{c}$ from the fake image. But we can not make sure which vector value will correspond to which emotion in our design, **G** will randomly build a vector value to emotion mapping.

"If I collect 100 smiling face and input to Encoder, will these 100 latent vectors be a cluster in high dimension space?"
Yes, if we assume 100 smiling corresponding to 100 vectors, those vectors should be very close to each other. 
I think continuity is some what a intrinsic characteristic of neural networks. Since we use continuous real numbers represent $\tilde{c}$, if two vectors are close to each other, it will have higher possibility to generate similar emotions compare to two vectors that far away from each other.

**Comment by ken881015** [comment]
Thank you for replying!!

- As you say, "**We can not make sure which vector value will correspond to which emotion in our design, G will randomly build a vector value to emotion mapping.**", then why "**if we assume 100 smiling corresponding to 100 vectors, those vectors should be very close to each other.**" hold true? I believe that it's the mystery of machine learning, haha.
